docs/mindsdb_sql/functions/custom_functions.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: Upload Custom Functions
+sidebarTitle: Custom Functions
+---
+
+Custom functions provide advanced means of manipulating data. Users can upload custom functions written in Python to MindsDB and apply them to data.
+
+## How It Works
+
+You can upload your custom functions via the MindsDB editor by clicking `Add` and `Upload custom functions`, like this:
+
+<p align="center">
+  <img src="/assets/upload_custom_function.png" />
+</p>
+
+Here is the form that needs to be filled out in order to bring your custom functions to MindsDB:
+
+<p align="center">
+  <img src="/assets/upload_custom_function_empty_form.png" />
+</p>
+
+Let's briefly go over the files that need to be uploaded:
+
+* The Python file stores an implementation of your custom functions. Here is the sample format:
```

Comment:
Lets highlight that input and output types annotation for function are important. If they are not set - default type 'str' will be used

Reply:
Done.